sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2268,6 +2269,18 @@ class Analyzer(
     }
   }
 
+  object ResolveAsofJoin extends Rule[LogicalPlan] {
+    override def apply(plan: LogicalPlan): LogicalPlan = plan match {
```

Comment:
Should probably use `resolveOperatorsUp` instead of `match` (Try to understand them)

Reply:
Leave a TODO here because I think match will break with "df.select(...).mergeAsof(df2, ...)"

Reply:
Resolved